src/hotspot/share/ci/ciInstanceKlass.cpp

```diff
@@ -729,6 +729,19 @@ void ciInstanceKlass::dump_replay_instanceKlass(outputStream* out, InstanceKlass
   }
 }
 
+GrowableArray<ciInstanceKlass*>* ciInstanceKlass::transitive_interfaces() const {
+  GrowableArray<ciInstanceKlass*>* result = NULL;
+  GUARDED_VM_ENTRY(
+    InstanceKlass* ik = get_instanceKlass();
```

Comment:
Maybe but it's not as simple as it seems (for the reason discussed below for vm classes). Some classes are allocated in a special arena because they are shared between compilations. A _transitive_interfaces array would need to be allocated in that same arena (otherwise if lazily allocated during a particular compilation it would be in the thread's resource are and  its content would be wiped out when the compilation is over but still reachable). That's not straightforward with the current implementation as the arena a ciInstanceKlass is allocated in is not available to the ciInstanceKlass so that would require extra changes and I'm not sure the extra complexity is worth it.

Reply:
scratch that. I see it's done for `_nonstatic_fields` and will do the same.